### DISCUSSION TITLE
Seeed Studios XIAO ESP32C6 Port

### DISCUSSION THREAD

**Discussion by BinTechLLC** [micropython/micropython#16489] [Core Development]
I am curious if there are plans to port an official build for the Seeed Studios XIAO ESP32C6 chip. I have tried multiple builds of similar boards that are on the downloads pages but none of them work. They have a MicroPython version that is buggy at best and is really hard to even keep the REPL connected. I gave up with how buggy it was and went back to C++ for now, but there are some things I can't seem to find a solution for in C++ that I know of multiple ways how to do in MicroPython. Would be awesome for a lot of other reasons too. :)

**Comment by sosi-deadeye** [comment]
What does not work?
I'm also using the XIAO ESP32C6 for testing purposes. (btw. the reset and boot buttons are incredible bad)
I could not recognize that something didn't work, but I wasn't testing every possible function.

One problem all ESP32 Microcontrollers have, are some special Pins which are used for SPIRAM and UART. If you set those Pins, the REPL would no longer work and if you do with a GPIO connected to the SPIRAM, the controller should crash.

I'm using always the latest Micropython-Version: https://micropython.org/download/esp32c6/

**Comment by BinTechLLC** [comment]
Just tried that one to see if I overlooked it and have the same issues. The primary issue is it just not responding when I hit the stop button or when I hit the play button while it is running. It just doesn't respond like all my other chips do. Usually I have to hit the stop button a bunch of times to get it to respond again.. The thing I can not seem to figure out how to get it to do in C++ is resolve a hostname to an IP address. I know multiple ways to do that in MicroPython or Python, but the ways I know how to do it in C++ do not seem to work. Now I can't remember what the error was it was giving me for that one in C++ either. The other MicroPython issue I had was it randomly disconnecting while its running. It would be running and printing stuff to the computer but then would just randomly stop and disconnect and I have to hit the reset button on the chip to get it to reset and reconnect. The project is basically a bunch of press buttons that send SMS alerts through a "remote" server. I have been using Socket to make these connections, which works fine, but when I test it on my hotspot so it isn't on the servers local network and uses the external address through the internet, it always resolves to 0.0.0.0 and I have force set the DNS to 8.8.8.8 and tried a bunch of different things and it just would not resolve, but my other chips running MicroPython can resolve it just fine over my hotspot. It seems like a really weird or even maybe a factory defect on this board from the factory.
